libnmstate/ifaces/base_iface.py

```diff
@@ -295,8 +295,9 @@ def master_type(self):
     def gen_metadata(self, ifaces):
         if self.is_master and not self.is_absent:
             for slave_name in self.slaves:
-                slave_iface = ifaces[slave_name]
-                slave_iface.set_master(self.name, self.type)
+                if slave_name != self.name:
```

Comment:
The else case should probable log a warning to make the problematic state more visible.

Reply:
Done!